src/System.Net.Requests/src/System/Net/FileWebRequest.cs

```diff
@@ -21,8 +21,8 @@ public class FileWebRequest : WebRequest, ISerializable
         private long _contentLength;
         private int _timeout = DefaultTimeoutMilliseconds;
         private bool _readPending;
-        private bool _writePending;
-        private bool _writing;
+        private volatile bool _writePending;
+        private volatile bool _writing;
```

Comment:
Why does _writing need to be volatile?

Reply:
Due to this result here. https://lgtm.com/projects/g/dotnet/corefx/snapshot/66eb2fbd4851008584c10a12d347ed8fc286bec6/files/src/System.Net.Requests/src/System/Net/FileWebRequest.cs?sort=name&dir=ASC&mode=heatmap#x5eef38a43faf1192:1

Reply:
Thanks, but that doesn't answer my question. What race condition is it preventing?

Reply:
Ha, you are right. `_writing` isn't updated in the critical region.